compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -420,8 +447,7 @@ macro_rules! define_queries {
                     hash_result: hash_result!([$($modifiers)*]),
                     handle_cycle_error: |tcx, mut error| handle_cycle_error!([$($modifiers)*][tcx, error]),
                     compute,
-                    cache_on_disk,
-                    try_load_from_disk: Self::TRY_LOAD_FROM_DISK,
+                    try_load_from_disk: if cache_on_disk { Self::TRY_LOAD_FROM_DISK } else { None },
```

Comment:
I think we can even put `should_ever_cache_on_disk!([$($modifiers)*])` here, and remove the `TRY_LOAD_FROM_DISK` associated constant.

Reply:
great idea, done :)